### ISSUE TITLE
Add ability to tag individual scan results

### ISSUE THREAD

**Issue by sdupreez** [natlas/natlas#207]
**Is your feature request related to a problem?**
After adding a few IP ranges into the scope section, my agents graciously start sending over results. I then, curiously start poking around what was found. What ports are open, in which ranges, and find some interesting trends and patterns. I then have to go and document some of these interesting things elsewhere. If we had the ability to add ad-hoc tags to individual scan results, it would greatly help me in finding things again at a later stage.

**Describe the feature you'd like**
The ability to add tags to scan results directly. Not only to scope elements like you can do currently.

**Have you considered alternative ways to get this feature**
Short of documenting patterns and / or trends somewhere else, no

**Additional context**
Currently scanning multiple ranges within a multi national organization, adding tags for countries, and looking for ways to organize things a bit more. Eg. tagging printers, switches, web servers, etc.

**Comment by 0xdade** [comment]
This could be done with a bit of rework to how tagging is done, or a new type of tagging. Would it also be valuable to be able to tag an individual host such that all scan results for that host always have a specific tag, but without having to create that host as an individual scope item? I think so, but it does also add an extra layer of potential complications with tagging.

Thanks for this submission, I think it's a good idea and I originally scrapped it because of risks of making a complicated UX.

**Comment by sdupreez** [comment]
> Would it also be valuable to be able to tag an individual host such that all scan results for that host always have a specific tag, but without having to create that host as an individual scope item?

Yes, this is actually the behavior that I am looking for. No need to tag individual ports, but rather the entire host, but not at the scope level.
